Dockerfile

```diff
@@ -0,0 +1,167 @@
+FROM continuumio/miniconda3:24.7.1-0
+
+# Argument to choose the model: piper, vits, xtts
+ARG TTS_MODEL="xtts"
+ENV TTS_MODEL=$TTS_MODEL
+
+ARG CUDA_VERSION="12.1.1"
+ENV CUDA_VERSION=$CUDA_VERSION
+
+ARG PYTHON_VERSION=3.11.9
+ENV PYTHON_VERSION=$PYTHON_VERSION
+
+ARG PYTORCH_VERSION=2.2.1
+ENV PYTORCH_VERSION=$PYTORCH_VERSION
+
+SHELL ["/bin/bash", "-l", "-c"]
+ENV SHELL=/bin/bash
+ENV HOST=0.0.0.0
+ENV DEBIAN_FRONTEND=noninteractive
+ENV CUDA_DOCKER_ARCH=all
+ENV GRADIO_SERVER_NAME="0.0.0.0"
+ENV NVIDIA_VISIBLE_DEVICES=all
+
+RUN <<EOR
+    apt-get update
+    apt-get upgrade -y
+    apt-get install --no-install-recommends -y \
+      espeak-ng \
+      curl \
+      wget \
+      jq \
+      vim
+
+    apt-get clean && rm -rf /var/lib/apt/lists/*
+EOR
+
+WORKDIR /alltalk
+
+# Create a conda environment and install dependencies:
+ARG INSTALL_ENV_DIR=/alltalk/alltalk_environment/env
+ENV CONDA_AUTO_UPDATE_CONDA="false"
+RUN <<EOR
+    CUDA_SHORT_VERSION=${CUDA_VERSION%.*}
+
+    conda create -y -n "alltalk" -c conda-forge python=${PYTHON_VERSION}
+    conda activate alltalk
+    RESULT=$( { conda install -y \
+      gcc_linux-64 \
+      gxx_linux-64 \
+      pytorch=${PYTORCH_VERSION} \
+      pytorch-cuda=${CUDA_SHORT_VERSION} \
+      torchvision \
+      torchaudio \
+      libaio \
+      nvidia/label/cuda-${CUDA_SHORT_VERSION}.0::cuda-toolkit \
+      faiss-gpu=1.9.0 \
+      conda-forge::ffmpeg=7.1.0 \
+      conda-forge::portaudio=19.7.0 \
+      -c pytorch \
+      -c anaconda \
+      -c nvidia ; } 2>&1 )
+
+    if echo $RESULT | grep -izq error ; then
+      echo "Failed to install conda dependencies 2: $RESULT"
+      exit 1
+    fi
+    conda clean -a && pip cache purge
+EOR
+
+# Install python dependencies (cannot use --no-deps because requirements are not complete)
+COPY system/config system/config
+COPY system/requirements/requirements_standalone.txt system/requirements/requirements_standalone.txt
+COPY system/requirements/requirements_parler.txt system/requirements/requirements_parler.txt
+ENV PIP_CACHE_DIR=/alltalk/pip_cache
+RUN <<EOR
+    conda activate alltalk
+
+    mkdir /alltalk/pip_cache
+    pip install --no-cache-dir --cache-dir=/alltalk/pip_cache -r system/requirements/requirements_standalone.txt
+    pip install --no-cache-dir --cache-dir=/alltalk/pip_cache --upgrade gradio==4.32.2
```

Comment:
@erew123 : I took this from https://github.com/erew123/alltalk_tts/blob/alltalkbeta/atsetup.sh#L361 

But why is this necessary?

Gradio is already part of requirements_standalone.txt. Isn't that enough?